<launch>
  <arg name="world_file"  default="$(find l5_robots)/maps/labyrinth.world"/>

  <arg name="base"      value="$(optenv TURTLEBOT_BASE kobuki)"/> <!-- create, roomba -->
  <arg name="battery"   value="$(optenv TURTLEBOT_BATTERY /proc/acpi/battery/BAT0)"/>  <!-- /proc/acpi/battery/BAT0 --> 
  <arg name="gui" default="true"/>
  <arg name="stacks"    value="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons --> 
  <arg name="3d_sensor" value="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro --> 

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="gui" value="$(arg gui)" />
    <arg name="world_name" value="$(arg world_file)"/>
  </include>
  
  <include file="$(find l5_robots)/launch/spawn_robot.launch">
    <arg name="base" value="$(arg base)"/>
    <arg name="stacks" value="$(arg stacks)"/>
    <arg name="3d_sensor" value="$(arg 3d_sensor)"/>
    <arg name="x" value="-0.433787" />
    <arg name="y" value="8.403358" />
    <arg name="z" value="-0.000248" />
    <arg name="roll" value="-0.000248" />
    <arg name="pitch" value="-0.000248" />
    <arg name="yaw" value="-1.586427" />
  </include>
  
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>
  
  <!-- Fake laser -->
  <node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
        args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
    <param name="scan_height" value="10"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.0"/>
    <remap from="image" to="/camera/depth/image_raw"/>
    <remap from="scan" to="/scan"/>
  </node>



  <param name="publish_frequency_hz" value="20" />

  <rosparam param="ROIl">[[20, 10], [10, 10]]</rosparam>
  <rosparam param="ROIr">[[20, 10], [30, 10]]</rosparam>
  <rosparam param="colour_mappings">  
      '0': (0, 0, 0) 
      '1': (255, 255, 255) 
      '2': (0, 150, 255)
  </rosparam>



  <!-- include file="$(find turtlebot_teleop)/launch/keyboard_teleop.launch" /-->

  <!--node pkg="rviz" type="rviz" name="rviz" args="-d $(find l5_robots)/rviz/single_robot.rviz" /-->

  <node pkg="l5_robots" name="Laser2CA" type="Laser2CA.py" output="screen">
    <param name="LASER_RANGE_MAX" value="5.0"/>
  </node>

  <node pkg="l5_robots" name="CA" type="CA.py" output="screen">
    <param name="HEIGHT" value="50"/>
    <param name="WIDTH" value="50"/>
    <param name="LIVE" value="1"/>
    <param name="DEAD" value="0"/>
    <param name="OBSTACLE" value="2"/>
  </node>

  <node pkg="l5_robots" name="CAVisualiser" type="CAVisualiser.py" output="screen">
  </node>

  <node pkg="l5_robots" name="CAInterpreter" type="CAInterpreter.py" output="screen">
    <param name="THRESH_LIMIT" value="0.0"/>
    <param name="REFRESH_LIMIT" value="300"/>
    <param name="RECOV_LIMIT" value="50"/>
  </node>

  <node pkg="l5_robots" name="SubsumptionArchitecture" type="SubsumptionArchitecture.py" output="screen"/>

  
</launch>